src/drive/web/modules/services/components/SuggestionProvider.jsx

```diff
@@ -46,44 +47,41 @@ class SuggestionProvider extends React.Component {
     )
   }
 
-  // fetches pretty much all the files and preloads FuzzyPathSearch
+  // fetches pretty much all the files not trashed and preloads FuzzyPathSearch
   async indexFiles() {
     const { client } = this.props
-    // TODO: fix me
-    // eslint-disable-next-line no-async-promise-executor
-    return new Promise(async resolve => {
-      const resp = await cozy.client.fetchJSON(
-        'GET',
-        `/data/io.cozy.files/_all_docs?include_docs=true`
-      )
-      const files = resp.rows
-        // TODO: fix me
-        // eslint-disable-next-line no-prototype-builtins
-        .filter(row => !row.doc.hasOwnProperty('views'))
-        .map(row => ({ id: row.id, ...row.doc }))
-
-      const folders = files.filter(file => file.type === TYPE_DIRECTORY)
-
-      const notInTrash = file =>
-        !file.trashed && !/^\/\.cozy_trash/.test(file.path)
-      const notOrphans = file =>
-        folders.find(folder => folder._id === file.dir_id) !== undefined
-
-      const normalizedFilesPrevious = files
-        .filter(notInTrash)
-        .filter(notOrphans)
-
-      const normalizedFiles = await Promise.all(
-        normalizedFilesPrevious.map(
-          async file =>
-            await makeNormalizedFile(client, folders, file, getIconUrl)
-        )
+    this.hasIndexFilesBeenLaunched = true
+
+    let next = true
+    let files = []
+    let bookmark = ''
+
+    while (next) {
+      const { selector, options } = prepareSuggestionQuery(bookmark)
```

Comment:
nit: since the selector does not change, it is not ideal to recompute it at each while iteration, only the bookmark needs to change

Reply:
This method is not called several time.